packages/idos-sdk-js/src/lib/auth.ts

```diff
@@ -171,8 +177,24 @@ export class Auth {
       );
     };
 
+    const { connect } = nearAPI;
+    const connectionConfig = {
+      networkId: import.meta.env.VITE_IDOS_NEAR_DEFAULT_NETWORK,
+      nodeUrl: import.meta.env.VITE_IDOS_NEAR_DEFAULT_RPC_URL,
+    };
+    const nearConnection = await connect(connectionConfig);
+
+    const response = await nearConnection.connection.provider.query({
+      request_type: "view_access_key_list",
+      finality: "final",
+      account_id: currentAddress,
+    });
+    const public_key_with_prefix = response.keys.find((element: object) => element.access_key.permission == "FullAccess").public_key;
+    const public_key = public_key_with_prefix.replace(/^ed25519:/, "");
+    const implicitAccount = toBeHex(decodeBase58(public_key));
+
     return this.#setSigner({
-      accountId: currentAddress,
+      accountId: implicitAccount,
```

Comment:
Are you folks sure? Won't this mean that people will stop seeing their account ids on the dahsboard (and other places)? I'm not super sure how I feel about that.